RadixWallet/Features/TransactionReviewFeature/TransactionReviewAccount/TransactionReviewAccount+View.swift

```diff
@@ -96,134 +93,64 @@ extension TransactionReviewAccount {
-	init(resource: OnLedgerEntity.Resource, details: TransactionReview.Transfer.Details.Fungible) {
-		self.init(
-			name: resource.metadata.title,
-			thumbnail: .token(details.isXRD ? .xrd : .other(resource.metadata.iconURL)),
-			amount: details.amount,
-			guaranteedAmount: details.guarantee?.amount,
-			fiatAmount: nil
-		)
-	}
-}
-
-extension TransferNFTView.ViewState {
-	init(resource: OnLedgerEntity.Resource, details: TransactionReview.Transfer.Details.NonFungible) {
-		self.init(
-			tokenID: details.id.localId().toUserFacingString(),
-			tokenName: details.data?.name,
-			thumbnail: resource.metadata.iconURL
-		)
+		let nonXrdResources = resources.nonXrdResources.map {
+			Element(resourceWithRedemptionValue: $0, isXRD: false)
+		}
+		self = (xrdResource.map { [$0] } ?? []) + nonXrdResources
 	}
 }
 
-extension PoolUnitView.ViewState {
-	init(resource: OnLedgerEntity.Resource, details: TransactionReview.Transfer.Details.PoolUnit) {
+extension ResourceBalance.ViewState.Fungible { // FIXME: GK use full
+	init(resourceWithRedemptionValue resource: OnLedgerEntitiesClient.OwnedResourcePoolDetails.ResourceWithRedemptionValue, isXRD: Bool) {
 		self.init(
-			poolName: resource.fungibleResourceName,
-			amount: details.details.poolUnitResource.amount,
-			guaranteedAmount: details.guarantee?.amount,
-			dAppName: .success(details.details.dAppName),
-			poolIcon: resource.metadata.iconURL,
-			resources: .success(.init(resources: details.details)),
-			isSelected: nil
+			address: resource.resource.resourceAddress,
+			icon: .token(isXRD ? .xrd : .other(resource.resource.metadata.iconURL)),
+			title: isXRD ? Constants.xrdTokenName : resource.resource.metadata.title,
+			amount: resource.redemptionValue.map { .init($0) }
 		)
 	}
 }
 
-extension [PoolUnitResourceView.ViewState] {
+extension [ResourceBalance.Fungible] {
 	init(resources: OnLedgerEntitiesClient.OwnedResourcePoolDetails) {
 		let xrdResource = resources.xrdResource.map {
-			PoolUnitResourceView.ViewState(resourceWithRedemptionValue: $0, isXRD: true)
+			Element(resourceWithRedemptionValue: $0, isXRD: true)
 		}
 		let nonXrdResources = resources.nonXrdResources.map {
-			PoolUnitResourceView.ViewState(resourceWithRedemptionValue: $0, isXRD: false)
+			Element(resourceWithRedemptionValue: $0, isXRD: false)
 		}
-
 		self = (xrdResource.map { [$0] } ?? []) + nonXrdResources
 	}
 }
 
-extension PoolUnitResourceView.ViewState {
+extension ResourceBalance.Fungible {
 	init(resourceWithRedemptionValue resource: OnLedgerEntitiesClient.OwnedResourcePoolDetails.ResourceWithRedemptionValue, isXRD: Bool) {
 		self.init(
-			id: resource.resource.id,
-			symbol: isXRD ? Constants.xrdTokenName : resource.resource.metadata.title,
-			icon: .token(isXRD ? .xrd : .other(resource.resource.metadata.iconURL)),
-			amount: resource.redemptionValue
+			isXRD: isXRD,
+			amount: resource.redemptionValue ?? 0, // FIXME: GK - best way to handle nil amount?
```

Comment:
FIXME comment still relevant?

Reply:
Yes this one is, I am not sure if this is the best way to handle it, perhaps those pool unit resources should rather be filtered out